qiskit_machine_learning/circuit/library/qnn_circuit.py

```diff
@@ -0,0 +1,188 @@
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The QNN circuit."""
+
+from typing import Optional, List
+from qiskit.circuit import QuantumRegister, QuantumCircuit
+from qiskit.circuit.parametertable import ParameterView
+from qiskit.circuit.library import BlueprintCircuit
+from qiskit_machine_learning import QiskitMachineLearningError
+from qiskit_machine_learning.utils import derive_num_qubits_feature_map_ansatz
+
+
+class QNNCircuit(BlueprintCircuit):
+    """
+    The QNN circuit is a blueprint circuit that wraps feature map and ansatz circuits.
+    It can be used to simplify the composition of these two.
+
+    If only the number of qubits is provided the ``RealAmplitudes`` ansatz ``ZZFeatureMap``
+    are used. If the number of qubits is 1 the ``ZFeatureMap``is used.
+    If only a feature map is provided, the ``RealAmplitudes``ansatz with the
+    corresponding number of qubits is used. If only an ansatz is provided the ``ZZFeatureMap``
+    with the corresponding number of qubits is used.
+
+    At least one parameter has to be provided. If a feature map and an ansatz is provided, the
+    number of qubits must be the same.
+
+    Example:
+
+    .. code-block::
+        from qiskit_machine_learning.circuit.library import QNNCircuit
+        qnn_qc = QNNCircuit(2)
+        print(qnn_qc)
+        # prints:
+        #      ┌──────────────────────────┐»
+        # q_0: ┤0                         ├»
+        #      │  ZZFeatureMap(x[0],x[1]) │»
+        # q_1: ┤1                         ├»
+        #      └──────────────────────────┘»
+        # «     ┌──────────────────────────────────────────────────────────┐
+        # «q_0: ┤0                                                         ├
+        # «     │  RealAmplitudes(θ[0],θ[1],θ[2],θ[3],θ[4],θ[5],θ[6],θ[7]) │
+        # «q_1: ┤1                                                         ├
+        # «     └──────────────────────────────────────────────────────────┘
+
+        print(qnn_qc.num_qubits)
+        # prints: 2
+
+        print(qnn_qc.input_parameters)
+        # prints: ParameterView([ParameterVectorElement(x[0]), ParameterVectorElement(x[1])])
+
+        print(qnn_qc.weight_parameters)
+        # prints: ParameterView([ParameterVectorElement(θ[0]), ParameterVectorElement(θ[1]),
+        #         ParameterVectorElement(θ[2]), ParameterVectorElement(θ[3]),
+        #         ParameterVectorElement(θ[4]), ParameterVectorElement(θ[5]),
+        #         ParameterVectorElement(θ[6]), ParameterVectorElement(θ[7])])
+    """
+
+    def __init__(
+        self,
+        num_qubits: Optional[int] = None,
+        feature_map: Optional[QuantumCircuit] = None,
+        ansatz: Optional[QuantumCircuit] = None,
```

Comment:
According to the user's choice a valid configuration for number of qubits, feature map and ansatz is derived when the circuit is built. Thus, when the circuit is built, the feature map and ansatz are never None. In case no input is provided a QiskitMachineLearningError when attempting to build the circuit, see ``derive_num_qubits_feature_map_ansatz``.  If desired, the input can already be validated when being constructed with the same methodology as when being built. Let me know ;).